indy_node/test/auth_rule_change/test_constraint_field.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+
+from indy_common.constants import CONSTRAINT, AUTH_ACTION
+from indy_common.types import ClientAuthRuleOperation
+from indy_node.test.auth_rule_change.helper import generate_constraint_entity, generate_constraint_list, \
+    generate_auth_rule_operation
+from indy_common.authorize.auth_constraints import ROLE, AUTH_CONSTRAINTS, CONSTRAINT_ID
+
+validator = ClientAuthRuleOperation()
+
+valid_auth_rule_operation_small = generate_auth_rule_operation()
+valid_auth_rule_operation_large = generate_auth_rule_operation(constraint=generate_constraint_list(
+    auth_constraints=[generate_constraint_entity(),
+                      generate_constraint_entity()]))
+valid_auth_rule_operation_extra_large = generate_auth_rule_operation(constraint=generate_constraint_list(
+    auth_constraints=[generate_constraint_entity(),
+                      generate_constraint_list(
+                          auth_constraints=[generate_constraint_entity(),
+                                            generate_constraint_entity()])]))
+
+
+def test_valid():
+    validator.validate(valid_auth_rule_operation_small)
+    validator.validate(valid_auth_rule_operation_large)
+    validator.validate(valid_auth_rule_operation_extra_large)
+
+
+def test_invalid_operation_action():
+    # must be ADD_PREFIX or EDIT_PREFIX
+    invalid_auth_rule_operation = generate_auth_rule_operation(auth_action="auth_action")
+    with pytest.raises(TypeError) as e:
+        validator.validate(invalid_auth_rule_operation)
+    e.match(AUTH_ACTION)
+
+
+def test_invalid_entity_role():
+    # ConstraintEntityField without required field 'role'
+    invalid_auth_rule_operation = generate_auth_rule_operation()
+    del invalid_auth_rule_operation[CONSTRAINT][ROLE]
```

Comment:
Please add a test for validation of constraint in extra_large case (remove role from a second-level consraint)